src/helpers/buildHyperlinkHelper.ts

```diff
@@ -1,29 +1,55 @@
-import type { SelectedGeographyData } from "../types";
-import { areAllUndefined } from "../util/genUtil";
+interface TopicPageParams {
+  topic: string;
+}
 
-export const buildHyperlink = (args: {
-  topic?: string;
-  variable?: string;
+interface VariablePageParams {
+  topic: string;
+  variable: string;
+}
+
+interface CategoryPageParams {
+  topic: string;
+  variable: string;
   classification?: string;
-  category?: string;
-  selectedGeography?: SelectedGeographyData;
-}) => {
-  let url = "/";
-  let paramsArr = [args.topic, args.variable, args.classification, args.category];
-  if (args.topic) {
-    if (args.category && !args.classification) {
-      paramsArr[2] = "default";
+  category: string;
+}
+
+type UrlParams = TopicPageParams | VariablePageParams | CategoryPageParams;
+
+/**
+ * Function takes in current url and (optionally)
+ * topic / variable / classification / category in
+ * urlParams object or (optionally) a static path e.g. "topics"
+ * and returns complete hyperlink string.
+ * Omitting the urlParams parameter will return a link
+ * to the index page.
+ */
+export const buildHyperlink = (url: URL, urlParams?: UrlParams, staticPath?: string) => {
+  if (!urlParams && !staticPath) {
+    return `/${url.search}`;
+  } else if (staticPath) {
+    return `/2021/${staticPath}${url.search}`;
+  } else {
+    let link = "/2021";
+    if ("path" in urlParams) {
```

Comment:
are all of these if checks mutually exclusive? (i.e. would you ever get "path" and "topic" in the same urlParams?)